File: client/lib/src/domain/entities/query_element.dart
import 'package:equatable/equatable.dart';

import 'package:query_wizard/domain.dart';

//ignore: must_be_immutable
class QueryElement extends Equatable implements IEntity {
  QueryElement({
    required this.id,
    required this.name,
    this.alias,
    required this.type,
    this.parent,
    required this.elements,
  });

  @override
  final String id;
  final String name;
  String? alias;
  final QueryElementType type;
  final QueryElement? parent;
  final List<QueryElement> elements;

  @override
  List<Object> get props => [
        id,
        name,
        alias ?? '',
        type,
        parent ?? '',
        elements,
      ];
}

File: server/src/QueryWizard.Domain/src/QueryElement.cs
using System.Collections.Generic;

namespace QueryWizard.Domain
{
    public class QueryElement
    {
        public string Id { get; set; }

        public string Name { get; set; }

        public string Alias { get; set; }

        public QueryElementType Type { get; set; }

        public QueryElement Parent { get; set; }

        public List<QueryElement> Elements { get; set; }
    }
}
